docs/aggregators.rst

```diff
@@ -11,35 +13,85 @@ During a training session, the combiners will instantiate an Aggregator and use
    :width: 100%
    :align: center
 
-The above figure illustrates the overall flow. When a client completes a model update, the model parameters are streamed to the combiner, and a model update message is sent. The model parameters are written to file on disk, and the model update message is passed to a callback function, on_model_update. The callback function validates the model update, and if successful, puts the update message on an aggregation queue. The model parameters are written to disk at a configurable storage location at the combiner. This is done to avoid exhausting RAM memory at the combiner. As multiple clients send updates, the aggregation queue builds up, and when a certain criteria is met, another method, combine_models, starts processing the queue, aggregating models according to the specifics of the scheme (FedAvg, FedAdam, etc). 
+The figure above illustrates the overall workflow. When a client completes a model update, the model parameters are streamed to the combiner, 
+and a model update message is sent. The parameters are saved to a file on disk, and the update message is passed to a callback function, ``on_model_update``. 
+This function validates the model update and, if successful, places the update message in an aggregation queue. 
+The model parameters are saved to disk at a configurable storage location within the combiner to prevent exhausting RAM. 
+As multiple clients submit updates, the aggregation queue accumulates. Once specific criteria are met, another method, ``combine_models``, 
+begins processing the queue, aggregating models according to the specifics of the scheme (e.g., FedAvg, FedAdam).
 
-The user can configure several parameters that guide general behavior of the aggregation flow: 
+
+Using different aggregators
+----------------------------
+
+FEDn supports the following aggregation algorithms: 
+
+- FedAvg (default)
+- FedAdam 
+- FedYogi
+- FedAdaGrad
+
+The implementation of the methods from the FedOpt family follows the implemenation in this paper: https://arxiv.org/pdf/2003.00295.pdf 
+
+
+Training sessions can be configured to use a given aggregator method. For example, to use FedAdam:
+
+.. code:: python
+
+   session_config = {
+                    "helper": "numpyhelper",
```

Comment:
this is so confusing, is it called fedadam or fedopt?

Reply:
The aggregator is called "fedopt", and it can be configured to have different server-side momentum terms, leading to adam, yogi and adagrad. We could duplicate some common code and make three seprate plugins if we think that is more logical?

Reply:
I will leave it like this for now, but I agree that it could be more logical if refactored into individual files. We can consider it later.